src/AppInstallerRepositoryCore/Microsoft/Schema/1_4/Interface_1_4.cpp

```diff
@@ -66,6 +67,18 @@ namespace AppInstaller::Repository::Microsoft::Schema::V1_4
         savepoint.Commit();
     }
 
+    bool Interface::NotNeeded(const SQLite::Connection& connection, std::string_view tableName, std::string_view valueName, SQLite::rowid_t id) const
+    {
+        bool result = V1_0::Interface::NotNeeded(connection, tableName, valueName, id);
+
+        if (tableName == Schema::V1_0::VersionTable::TableName())
```

Comment:
I would put the filtering responsibility on the `DependenciesTable` as it is the one to know what it references.  In addition, it knows the name of the column; copying the string out to `Interface` is not good.